### PR TITLE
Fix windows tests

### LINKED ISSUE #485: Many tests failing under Windows.
A lot of failures appear to be due to hardcoded paths into places like `/tmp` which should be easy to fix. Other tests appear to fail due to unexpected line endings - `\r\n` vs `\n` - which I think suggests that we're testing what comes out of a zip file (`\n`) with a text file that has been auto-converted to `\r\n` in the checkout process?

### PR DESCRIPTION
**Pull request by jspanjers**
Fixes #485

### CODE REVIEW

**Comment by jspanjers** [comment]
It seems most of the Coveralls test fails with a decreased coverage of 0.0%.
However, when I look at the Coveralls details, I do not notice any decreased coverage.
Maybe this is a rounding error?

Would it be  possible to change the 'Coverage Decrease Threshold for failure' on https://coveralls.io/github/rubyzip/rubyzip/settings to something like, 0.1% ?

**Review by hainesr — COMMENTED** [review]
Thank you so much for this PR @jspanjers. Great work!

I just have a couple of questions in a couple of places.

**Comment by hainesr** [comment]
> It seems most of the Coveralls test fails with a decreased coverage of 0.0%.
However, when I look at the Coveralls details, I do not notice any decreased coverage.
Maybe this is a rounding error?

> Would it be possible to change the 'Coverage Decrease Threshold for failure' on https://coveralls.io/github/rubyzip/rubyzip/settings to something like, 0.1% ?

Yes, I hate Coveralls being so touchy by default, and I had no idea it could be adjusted! Also, while I like decent coverage, I'm not super bothered about it going up or down a bit when it's over 90% anyway. I'd be happy to have 90% as the threshold and then stop worrying about it. Unfortunately I don't have the required access to set this, as I'm not an owner of the rubyzip repo - maybe @jdleesmiller could set the threshold? Thanks!

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -17,6 +17,10 @@ jobs:
     runs-on: ${{ matrix.os }}-latest
     continue-on-error: ${{ endsWith(matrix.ruby, 'head') || matrix.os == 'windows' }}
     steps:
+    - name: Don't automatically add carriage returns to text files on checkout
+      if: matrix.os == 'windows'
+      run: git config --global core.autocrlf false
+
```
[the remark this answers is not in the document]
Comment (jspanjers) [reply .github/workflows/tests.yml:23]: Great suggestion! I will adjust my commit.

File: test/file_test.rb [hunk 2]
```diff
@@ -285,16 +285,22 @@ def test_add_stored
   end
 
   def test_recover_permissions_after_add_files_to_archive
+    if Zip::RUNNING_ON_WINDOWS
+      # Windows NT does not support granular permissions
+      return
+    end
```
[the remark this answers is not in the document]
Comment (jspanjers) [reply test/file_test.rb:291]: Ah yes! Skip seems more appropriate!

File: test/file_options_test.rb [hunk 3]
```diff
@@ -58,6 +58,10 @@ def test_restore_times_true
       zip.extract(ENTRY_2, EXTPATH_2)
     end
 
+    # this test is disabled on Windows for now, waiting for #486.
+    # please remove this after merging #486.
+    return if Zip::RUNNING_ON_WINDOWS
+
```
[the remark this answers is not in the document]
Comment (jspanjers) [reply test/file_options_test.rb:64]: Ah yes! Skip seems more appropriate!

**Comment by jspanjers** [comment]
I will make a new pull request in a minute.

**Pull request closed without merging**